tests/integration/questionnaire/test_questionnaire_last_viewed_guidance_hub.py

```diff
@@ -0,0 +1,226 @@
+        self.post(action="sign_out")
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # Then the last viewed guidance is shown
+        self.assertInUrl(
+            "/questionnaire/list-collector/?last_viewed_question_guidance=True"
+        )
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/household-interstitial/"
+        )
+
+    def test_last_viewed_question_guidance_shown_on_section_in_progress_resume_list_collector_add_person(
+        self
+    ):
+
+        # Given
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # When I sign out and resume on a list collector add person
+        self.post()
+        self.post_address_confirmation_answer()
+        self.post_you_live_here_answer()
+        self.post_primary_person_answer()
+        self.post({"anyone-else": "Yes"})
+        self.post(action="sign_out")
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # Then the last viewed guidance is shown on the parent page
+        self.assertInUrl(
+            "/questionnaire/list-collector/?last_viewed_question_guidance=True"
+        )
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/household-interstitial/"
+        )
+
+    def test_last_viewed_question_guidance_shown_on_section_in_progress_resume_relationships(
+        self
+    ):
+
+        # Given
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # When I sign out and resume on a relationship question
+        self.post()
+        self.post_address_confirmation_answer()
+        self.post_you_live_here_answer()
+        self.post_primary_person_answer()
+        self.post_list_collector_answers()
+        self.post()
+        self.post()
+        self.post(action="sign_out")
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # Then the last viewed guidance is shown
+        self.assertInUrl("/questionnaire/relationships/")
+        self.assertInUrl("?last_viewed_question_guidance=True")
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/relationship-interstitial/"
+        )
+
+    def last_viewed_question_guidance_shown_assertion(self, link):
+        selector = "#section-start-link"
+        selected = self.getHtmlSoup().select(selector)
+        self.assertInBody("This is the last viewed question in this section")
+        self.assertEqual(selected[0].get("href"), link)
+
+    def last_viewed_question_guidance_not_shown_assertion(self):
+        self.assertNotInUrl("last_viewed_question_guidance=True")
+        self.assertNotInBody("This is the last viewed question in this section")
```

Comment:
The method naming implies these methods do the opposite of each-other, but the first method also checks the link, and the second checks the url. It would be clearer if the link checking was in an assertion method of its own and the shown assertion included a check for the url.

Also, as these assertions are shared between the two tests we should consider how we can remove the duplication.